src/middlewares/patient.js

```diff
@@ -3,18 +3,26 @@
 
     var keyFor = {
         "Observation": "subject",
-        "MedicationPrescription": "patient"
+        "MedicationPrescription": "patient",
+        "Encounter": "patient"
```

Comment:
In DSTU2  every resource with a patient has a search parameter called "patient" (sometimes it's just an alias for "subject") so we don't need any special logic -- we can just automatically add  a "patient"value whenever this middleware is invoked; keyFor can go away.